### PR TITLE
Convert pool query to function call

### PR DESCRIPTION
**Pull request by pschorf**
## Changes proposed in this PR
- Convert pool query logic to pure function call

## Why are we making these changes?
- Easier to read
- Easier to reuse
- Better performance
```
New (no pool)
Evaluation count : 83280 in 60 samples of 1388 calls.
             Execution time mean : 724.906725 µs
    Execution time std-deviation : 7.126763 µs
   Execution time lower quantile : 717.701292 µs ( 2.5%)
   Execution time upper quantile : 740.208994 µs (97.5%)
                   Overhead used : 1.565043 ns

Found 3 outliers in 60 samples (5.0000 %)
	low-severe	 1 (1.6667 %)
	low-mild	 2 (3.3333 %)
 Variance from outliers : 1.6389 % Variance is slightly inflated by outliers


Old (no pool)
Evaluation count : 40980 in 60 samples of 683 calls.
             Execution time mean : 1.470291 ms
    Execution time std-deviation : 8.615948 µs
   Execution time lower quantile : 1.453152 ms ( 2.5%)
   Execution time upper quantile : 1.486338 ms (97.5%)
                   Overhead used : 1.565043 ns


New (has pool)
Evaluation count : 54660 in 60 samples of 911 calls.
             Execution time mean : 1.090170 ms
    Execution time std-deviation : 16.439733 µs
   Execution time lower quantile : 1.067333 ms ( 2.5%)
   Execution time upper quantile : 1.124302 ms (97.5%)
                   Overhead used : 1.565043 ns

Found 5 outliers in 60 samples (8.3333 %)
	low-severe	 4 (6.6667 %)
	low-mild	 1 (1.6667 %)
 Variance from outliers : 1.6389 % Variance is slightly inflated by outliers

Old (has pool)
Evaluation count : 35460 in 60 samples of 591 calls.
             Execution time mean : 1.754258 ms
    Execution time std-deviation : 51.372349 µs
   Execution time lower quantile : 1.706617 ms ( 2.5%)
   Execution time upper quantile : 1.871717 ms (97.5%)
                   Overhead used : 1.565043 ns

Found 9 outliers in 60 samples (15.0000 %)
	low-severe	 6 (10.0000 %)
	low-mild	 3 (5.0000 %)
 Variance from outliers : 15.8231 % Variance is moderately inflated by outliers
```

### CODE REVIEW

File: scheduler/src/cook/mesos/util.clj [hunk 1]
```diff
@@ -485,18 +486,7 @@
...
-                        (and
-                          [(true? ?requesting-default-pool)]
-                          [(missing? $ ?j :job/pool)])
-
-                        ; We should also include jobs with a pool specified that
-                        ; matches the pool that the caller is requesting
-                        (and
-                          [?j :job/pool ?p]
-                          [?p :pool/name ?pool-name]))]
+               [(cook.mesos.pool/check-pool $ ?j :job/pool ?pool-name ?requesting-default-pool)]]
```
Comment (dposada) [inline scheduler/src/cook/mesos/util.clj]: Can this just be `pool/check-pool`?

File: scheduler/src/cook/mesos/pool.clj [hunk 2]
```diff
@@ -1,6 +1,16 @@
 (ns cook.mesos.pool
   (:require [datomic.api :as d]))
 
+(defn check-pool
+  "Returns true if requesting-default-pool? and the entity does not have a pool
+   or the entity has a pool named pool-name"
+  [source eid relationship pool-name requesting-default-pool?]
```
Comment (dposada) [inline scheduler/src/cook/mesos/pool.clj]: Maybe `entity->pool-fn` instead of `relationship`?

Comment (pschorf) [reply scheduler/src/cook/mesos/pool.clj]: I like `entity->pool` if that works for you

File: scheduler/src/cook/mesos/util.clj (hunk 1 above)
Comment (pschorf) [reply scheduler/src/cook/mesos/util.clj]: Unfortunately not, it seems like when datomic `eval`-s the query, the namespace aliases aren't in scope. I removed `cook.mesos.pool` from the `:require` clause, though.

**Comment by dposada** [comment]
LGTM, will merge when the build is green

**Comment by pschorf** [comment]
Build is now green

**Pull request merged**